src/schedlib/policies/satp1.py

```diff
@@ -0,0 +1,234 @@
+            },
+            'mars': {
+                'type' : 'source',
+                'name' : 'mars',
+            }
+        },
+    }
+
+def make_operations(az_speed, az_accel, disable_hwp=False, apply_boresight_rot=True, hwp_cfg=None):
+    if hwp_cfg is None:
+        hwp_cfg = { 'iboot2': 'power-iboot-hwp-2', 'pid': 'hwp-pid', 'pmx': 'hwp-pmx', 'hwp-pmx': 'pmx', 'gripper': 'hwp-gripper' }
+    pre_session_ops = [
+        { 'name': 'sat.preamble'        , 'sched_mode': SchedMode.PreSession, 'hwp_cfg': hwp_cfg, },
+        { 'name': 'sat.ufm_relock'      , 'sched_mode': SchedMode.PreSession, },
+        { 'name': 'sat.set_scan_params' , 'sched_mode': SchedMode.PreSession, 'az_speed': az_speed, 'az_accel': az_accel, },
+        { 'name': 'sat.hwp_spin_up'     , 'sched_mode': SchedMode.PreSession, 'disable_hwp': disable_hwp, },
+    ]
+    post_session_ops = [
+        { 'name': 'sat.hwp_spin_down'   , 'sched_mode': SchedMode.PostSession, 'disable_hwp': disable_hwp, },
+        { 'name': 'sat.wrap_up'         , 'sched_mode': SchedMode.PostSession, 'az_stow': 180, 'el_stow': 48},
+    ]
+    cal_ops = [
+        { 'name': 'sat.hwp_spin_down'   , 'sched_mode': SchedMode.PreCal, 'disable_hwp': disable_hwp, },
+        { 'name': 'sat.det_setup'       , 'sched_mode': SchedMode.PreCal, 'disable_hwp': disable_hwp, },
+        { 'name': 'sat.setup_boresight' , 'sched_mode': SchedMode.PreCal, 'apply_boresight_rot': apply_boresight_rot, },
+        { 'name': 'sat.hwp_spin_up'     , 'sched_mode': SchedMode.PreCal, 'disable_hwp': disable_hwp},
+        { 'name': 'sat.source_scan'     , 'sched_mode': SchedMode.InCal },
+        { 'name': 'sat.bias_step'       , 'sched_mode': SchedMode.PostCal, 'indent': 4},
+    ]
+    cmb_ops = [
+        { 'name': 'sat.det_setup'       , 'sched_mode': SchedMode.PreObs, 'disable_hwp': disable_hwp, },
+        { 'name': 'sat.hwp_spin_up'     , 'sched_mode': SchedMode.PreObs, 'disable_hwp': disable_hwp, },
+        { 'name': 'sat.setup_boresight' , 'sched_mode': SchedMode.PreObs, 'apply_boresight_rot': apply_boresight_rot, },
+        { 'name': 'sat.bias_step'       , 'sched_mode': SchedMode.PreObs, },
+        { 'name': 'sat.cmb_scan'        , 'sched_mode': SchedMode.InObs, },
+        { 'name': 'sat.bias_step'       , 'sched_mode': SchedMode.PostObs, },
+    ]
+    return pre_session_ops + cal_ops + cmb_ops + post_session_ops
+
+def make_config(
+    master_file,
+    az_speed,
+    az_accel,
+    cal_targets,
+    **op_cfg
+):
+    blocks = make_blocks(master_file)
+    geometries = make_geometry()
+    operations = make_operations(az_speed, az_accel, **op_cfg)
+
+    config = {
+        'blocks': blocks,
+        'geometries': geometries,
+        'rules': {
+            'sun-avoidance': {
+                'min_angle': 45,
+            },
+            'min-duration': {
+                'min_duration': 600
+            },
+        },
+        'operations': operations,
+        'allow_partial': False,
```

Comment:
it'd be nice to have all of these over-rideable on the creation. I noticed it while wanting to run SATP1Policy.from_defaults but wanting a Jupiter scan. During these time periods Jupiter doesn't actually rise through any of the available "focuses" so it has to be run with allow partial = True. 

Actually... as I think about it. Letting us set allow partial = True for individual cal targets would be the nicest approach.